plugins/BTaggingExerciseII.cc

```diff
@@ -94,7 +96,12 @@ BTaggingExerciseII::BTaggingExerciseII(const edm::ParameterSet& iConfig) :
        bDiscr_flav = bDiscr + "_" + flav;
        if( bDiscr.find("Counting") != std::string::npos ) // track counting discriminator can be both positive and negative and covers a wider range then other discriminators
          bDiscriminatorsMap[bDiscr_flav] = fs->make<TH2F>(bDiscr_flav.c_str(), (bDiscr_flav + ";Jet p_{T} [GeV];b-tag discriminator").c_str(), 20, 0, 200, 11000, -15, 40);
-       else
+       else if ( bDiscr.find("probbb") != std::string::npos || bDiscr.find("probb") != std::string::npos ) {
```

Comment:
Don't you create the same histogram twice with the OR, I think that a single part e.g.
```
bDiscr.find("probbb") != std::string::npos
```
Should suffice and avoid potential memory leaks